paz/backend/image/image.py

```diff
@@ -351,3 +352,194 @@ def scale_resize(image, image_size):
     image_scale = np.array(1 / image_scale)
     output_image = output_image[np.newaxis]
     return output_image, image_scale
+
+
+def compute_resizing_shape(image, size):
+    """Computes the final size of the image to be scaled by `size`
+    such that the largest dimension of the image is equal to `size`.
+
+    # Arguments
+        image: Array, raw image to be scaled.
+        size: Int, final size of the image.
+
+    # Returns
+        List: Containing final shape of image and scale.
+    """
+    H, W = image.shape[:2]
+    image_scale = size / max(H, W)
+    resizing_W = int(W * image_scale)
+    resizing_H = int(H * image_scale)
+    resizing_shape = (resizing_W, resizing_H)
+    return [resizing_shape, np.array(image_scale)]
+
+
+def pad_image(image, size, mode):
+    """Pads the image to the final size `size`.
+
+    # Arguments
+        image: Array, image to be padded.
+        size: Int, final size of the image.
+        mode: Str, specifying the type of padding.
+
+    # Returns
+        Array: Padded image.
+    """
+    H, W = image.shape[:2]
+    pad_H = size - H
+    pad_W = size - W
+    pad_shape = [(0, pad_H), (0, pad_W), (0, 0)]
+    return np.pad(image, pad_shape, mode=mode)
+
+
+def auto_contrast(image):
+    """Performs autocontrast or automatic contrast enhancement in a
+    given image. This method achieves this by computing the image
+    histogram and removing a certain `cutoff` percent from the lighter
+    and darker part of the histogram and then stretching the histogram
+    such that the lightest pixel gray value becomes 255 and the darkest
+    ones become 0.
+
+    # Arguments
+        image: Array, raw image.
+
+    # Returns:
+        contrasted: Array, contrast enhanced image.
+
+    # References:
+        [Python Pillow autocontrast](
+            https://github.com/python-pillow/Pillow/blob/main'
```

Comment:
The paper uses autocontrast from PIL which has a default cutoff=0. Which theoretically is equivalent to histogram equalization. Since we have histogram equalization as a paz processor this Autocontrast computation is redundant. Hence this is now fully removed from paz.